### ISSUE TITLE
Validation for array returns error for only 1st item and ignores other errors

### ISSUE THREAD

**Issue by NehaThaker** [ctavan/express-validator#477]
Hi, 
I am using schema-validation with express-validator 4.3.0, using a jsonSchema.json file.
Within the schema, there is an array of objects. 
express-validator returns an error if only the 1st object value is incorrect. 
If other objects i.e 2nd onwards (index : 1 ) has incorrect values, it does not return errors.

Using code as :
```
  req.checkBody(jsonSchema);
  return req.asyncValidationErrors();
```

eg: jsonSchema
```
 {
"property1":  {
    "optional": true
  },
  "propertytest.*": {
    "isURL": true
  }
}
```

Data:
```
{
"property1":  {
  "test":"testvalue"
},
"propertytest": [
   "http://sldjfsdjf.sldfjsdf.com/sdlfjskd",
    "test"
]
}
```

I have looked at the code and it seems this is because in express_validator.js:
following code picks up only the first value from selectFields() as selectFields(req, context)[0]

```
return function(req, res, next) {
    const contexts = [];
    function runContexts() {
      contexts.filter(context => !context.promise).forEach(context => {
        const field = selectFields(req, context)[0];
```

Can you please let me know if there is a workaround for this and an example of it.

**Comment by NehaThaker** [comment]
Also, need all the errors back for all the values that are incorrect in an array.

**Comment by andhikamaheva** [comment]
Happened with me too 😳

**Comment by gustavohenke** [comment]
I honestly don't remember why it was done this way. Will debug it further.

**Comment by gustavohenke** [comment]
Potential fix with a7156181a5262342b7abdd1ba7cb80da85680bc3. Will let you guys know when it's published.

**Comment by andhikamaheva** [comment]
Thank you @gustavohenke 🙏
